subiquity/server/server.py

```diff
@@ -354,6 +353,11 @@ def __init__(self, opts, block_log_dir):
     def set_source_variant(self, variant):
         self.variant = variant
 
+        # Update the model variant too.
+        # Guard for init. The "base_model" attribute isn't set until .run() is called.
```

Comment:
Ugh that's a bit nasty.

Reply:
I agree and now I'm looking at it again, I think we could do this a bit smarter. The problem is I wanted to set the variant through this interface on `__init__` but the model hasn't been attached yet. This way also doesn't solve the problem of the informing the model about our variant when we attach it (though the way the code exists now it's not strictly necessary).

Reply:
I refactored this slightly and added some comments to `make_model` to hopefully make this a little less gross.